dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild

```diff
@@ -1,47 +1,85 @@
-# Copyright 1999-2021 Gentoo Authors
+# Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
 
-inherit cmake
+inherit cmake multibuild
 
 HOMEPAGE="https://github.com/frankosterfeld/qtkeychain"
 DESCRIPTION="Qt API for storing passwords securely"
 
 if [[ ${PV} != *9999* ]]; then
-	SRC_URI="https://github.com/frankosterfeld/qtkeychain/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	SRC_URI="https://github.com/frankosterfeld/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 	KEYWORDS="amd64 arm64 ~ppc64 ~riscv x86"
 else
 	inherit git-r3
-	EGIT_REPO_URI="${HOMEPAGE}.git"
+	EGIT_REPO_URI="https://github.com/frankosterfeld/${PN}.git"
 fi
 
 LICENSE="BSD-2"
 SLOT="0/1"
-IUSE="gnome-keyring"
+IUSE="gnome-keyring +qt5 qt6"
+REQUIRED_USE="|| ( qt5 qt6 )"
+
+# no tests
+RESTRICT="test"
 
 DEPEND="
-	dev-qt/qtcore:5
-	dev-qt/qtdbus:5
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtdbus:5
+	)
+	qt6? (
+		dev-qt/qtbase:6[dbus]
+	)
 	gnome-keyring? (
 		app-crypt/libsecret
 		dev-libs/glib:2
 	)
 "
 RDEPEND="${DEPEND}"
 BDEPEND="
-	dev-qt/linguist-tools:5
+	qt5? ( dev-qt/linguist-tools:5 )
+	qt6? ( dev-qt/qttools:6[linguist] )
 "
 
 DOCS=( ChangeLog ReadMe.md )
 
-src_configure() {
+pkg_setup() {
+	MULTIBUILD_VARIANTS=( $(usev qt5) $(usev qt6) )
+}
+
+with_qt6() {
+	case ${MULTIBUILD_VARIANT} in
+		qt5) echo OFF ;;
+		qt6) echo ON ;;
+	esac
+}
+
+qt_get_mkspecsdir() {
+	echo "${EPREFIX}"/usr/$(get_libdir)/${MULTIBUILD_VARIANT}/mkspecs
+}
+
+my_src_configure() {
```

Comment:
we usually make these as nested function within `src_configure`

Reply:
fixed